lib/internal/async_hooks.js

```diff
@@ -243,13 +245,84 @@ function restoreActiveHooks() {
   active_hooks.tmp_fields = null;
 }
 
+const triggerAsyncIdSymbol = Symbol('triggerAsyncId');
+const destroyedSymbol = Symbol('destroyed');
+
+function wrapPromise(promise, parent, silent) {
+  const asyncId = getOrSetAsyncId(promise);
+
+  if (parent) {
+    promise[triggerAsyncIdSymbol] = getOrSetAsyncId(parent);
+    promise.isChainedPromise = true;
+  } else {
+    promise[triggerAsyncIdSymbol] = getDefaultTriggerAsyncId();
+    promise.isChainedPromise = false;
+  }
+
+  if (!silent && initHooksExist()) {
+    emitInitScript(asyncId, 'PROMISE', promise[triggerAsyncIdSymbol], promise);
+  }
+
+  if (destroyHooksExist()) {
```

Comment:
Can this be ever `true`? I think FastPromiseHook implies that there is no destroy hook existing.